aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3450,7 +3456,7 @@ static void linalg_lstsq_out_info(
     bool compute_residuals = true;
     if (driver == "gelss" || driver == "gelsd") {
       if (input.dim() == 2) {
-        compute_residuals = (rank.item().toInt() == n);
+        compute_residuals = (rank.item().toDouble() == n);
```

Comment:
@lezcano 

So there is a problem where this would run into an overflow if left as `toInt()`. The reason is because rank will have non-sensical values if it goes through the SVD code path, since unlike `lstsq_kernel`, `svd_kernel` doesn't touch the rank tensors. Now with that being said, I don't like this as a work around. I'm hoping to get your input on how we should approach this situation.

I don't want to rework the entire function and add more cognitive load on you and other reviewers. Not without your opinion anyway.

Reply:
You can get the rank from the computation, but that would it would synchronise. Now, this is already synchronising, so I guess it's alright.